guide/docs/interactions/modals.mdx

```diff
@@ -0,0 +1,195 @@
+-   `timeout`: The time (in seconds) after which the modal is removed from cache. Defaults to **600 seconds**.
+-   `components`: The list of components to be displayed in the modal. Limited to **5 action rows**.
+
+:::caution
+
+Modals without timeouts are not currently supported. There is no event dispatched when the modal is closed by the user, and thus the modal would not be removed from cache without a timeout.
+
+:::
+
+These attributes can be used while creating a modal, as follows.
+
+```python title="modals.py"
+# Subclassing the modal.
+class MyModal(disnake.ui.Modal):
+    def __init__(self):
+        super().__init__(
+            title="Modal Title",
+            custom_id="modal_id",
+            timeout=300,
+            components=[...],
+        )
+
+
+# Using the lower-level interface. (inside a function)
+await inter.response.send_modal(
+    title="Modal Title",
+    custom_id="modal_id",
+    timeout=300,
+    components=[...],
+)
+```
+
+## `TextInput` components
+
+`TextInput` is a component in itself, like buttons and select menus. It is responsible for receiving user input in the form of text, and cannot be used outside of modals. It has the following customizable attributes.
+
+-   `label`: The label of the text input.
+-   `custom_id`: An ID specified to the text input.
+-   `style`: Utilizes one of <DocsLink reference="disnake.TextInputStyle" /> for the component.
+-   `placeholder`: The placeholder text that is shown if nothing is entered.
+-   `value`: The pre-filled value of the text input (upto character limit).
+-   `required`: Whether the text input is required. Defaults to `True`.
+-   `min_length` and `max_length`: Set the minimum and maximum length of the inputs respectively.
+
+These attributes can be used per `TextInput` component, as follows.
+
+```python title="textinput.py"
+components = [
+    disnake.ui.TextInput(
+        label="Name",
+        placeholder="Foo Tag",
+        custom_id="name",
+        style=TextInputStyle.short,
+        max_length=50,
+    ),
+]
+```
+
+The <DocsLink reference="disnake.TextInputStyle" /> attribute currently has two styles, which have multiple aliases:
+
+-   `short` - Represents a single-line text input component. Also called `single_line`.
+-   `long` - Represents a multi-line text input component. Also called `multi_line` or `paragraph`.
+
+## Modal coroutines
+
+### <DocsLink reference="disnake.ui.Modal.callback">`callback`</DocsLink>
+
+### <DocsLink reference="disnake.ui.Modal.on_error">`on_error`</DocsLink>
+
+### <DocsLink reference="disnake.ui.Modal.on_timeout">`on_timeout`</DocsLink>
+
+## Modal notes
+
+-   The <DocsLink reference="disnake.TextInputStyle" /> defaults to `short`, if not specified for the component.
+-   All <DocsLink reference="disnake.ui.TextInput">TextInput</DocsLink> components are required by default.
+
+## Modal limits
+
+-   A modal can have a total of **5 components**.
```

Comment:
```suggestion
-   A modal can have a total of **5 action rows**.
```
Also already mentioned above, but having all the limits in a separate section as well is a nice idea.